### ISSUE TITLE
Consider simplifying logically redundant clauses in expressions

### ISSUE THREAD

**Issue by pmonks** [pmonks/clj-spdx#35]
Currently, when a valid but logically redundant expression such as:
```clojure
"Apache-2.0 AND Apache-2.0"
```
is parsed, `spdx.expressions/parse` returns:
```clojure
[:and {:license-id "Apache-2.0"} {:license-id "Apache-2.0"}]
```
which is logically equivalent to:
```clojure
{:license-id "Apache-2.0"}
```
An implementation of this simplification should be considered, controllable via an option (e.g. `:collapse-redundant-clauses?`), and defaulting to `true` (enabled).

**Comment by pmonks** [comment]
Note that this will require recursive (depth-first) "simplification" logic to handle cases such as:
```clojure
"Apache-2.0 AND (Apache-2.0 OR (Apache-2.0 AND Apache-2.0))"
```
The result of parsing this monstrosity should be:
```clojure
{:license-id "Apache-2.0"}
```

**Comment by pmonks** [comment]
Fixed in PR #36.
